### PR TITLE
Feat/initialize  create UI

### PR DESCRIPTION
**Pull request by thomasdinh2k**
image

Created **UI Components** includes: 

- A Homepage with single navbar (able to toggle submenus)
- A Simple Login Page
- A Simple Sign Up Page

Review goal: 
- Make sure **file structure** is correct
- Make sure **component & state implementation** is correct

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,13 +1,16 @@
 import "./App.css"
+import "./components/common/styles.scss"
 import { Route, Routes } from "react-router-dom"
-import HomePage from "./components/HomePage/HomePage"
-import LoginScreen from "./components/LoginScreen/LoginScreen"
-import SettingScreen from "./components/SettingScreen/SettingScreen"
+import HomePage from "./components/pages/HomePage/Index"
+import LoginScreen from "./components/pages/LoginScreen/LoginScreen"
+import SettingScreen from "./components/pages/SettingScreen/SettingScreen"
+import SignUpScreen from "./components/pages/SignUpScreen/SignUpScreen"
```
Comment (nhcuongng) [inline src/App.jsx]: rename file to index.tsx (not uppercase I char)

File: src/App.jsx [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
 function App() {
 	return (
 		<>
 			<Routes>
 				<Route path="/" element={<LoginScreen />} />
+				<Route path="sign-up" element={<SignUpScreen />} />
```
Comment (nhcuongng) [inline src/App.jsx:7]: Insert "/" before "sign-up"

Comment (nhcuongng) [reply src/App.jsx:7]: Bonus: Move to constant
Example: 
const Routers = {
 path: '/',
 component: <Component />
 ...
}

File: src/components/common/Button.jsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from "react"
+
+export default function Button({type, children}) {
```
Comment (nhcuongng) [inline src/components/common/Button.jsx]: encapsulate file scss

File: src/components/common/styles.scss [hunk 4]
```diff
@@ -0,0 +1,171 @@
+$theme-orange: linear-gradient(to right, #ff512f 0%, #f09819 51%, #ff512f 100%);
+$theme-orange-dark: linear-gradient(
+	to right,
+	#e04627 0%,
+	#cc7a15 51%,
+	#e04627 100%
+);
+$theme-carrot: #f28d1c;
```
Comment (nhcuongng) [inline src/components/common/styles.scss]: these vars move to a single file like, var.scss. And move it to styles folder at root

File: src/components/pages/HomePage/NavBar/NavBar.scss [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+	.head {
+		height: 120px;
+		padding: 30px;
+	}
+
+	nav {
+		display: none;
+		position: relative;
+		top: 0;
+		min-height: 100svh;
```
Comment (nhcuongng) [inline src/components/pages/HomePage/NavBar/NavBar.scss]: Move to vh

File: src/components/pages/HomePage/NavBar/NavBar.scss [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+		gap: 10px 10px;
+		/* padding-block: 2rem; */
+		/* font-size: 2rem; */
+
+		margin: 0;
+		list-style: none;
+	}
+
+	.nav-list li {
+		padding: 0.8rem 0.3rem 0.7rem 0.3rem;
```
Comment (nhcuongng) [inline src/components/pages/HomePage/NavBar/NavBar.scss]: change to utilites class, or change to pixel

File: src/components/pages/HomePage/NavBar/NavBar.scss [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+	}
+
+	.nav__list-text {
+		flex-grow: 1;
+	}
+
+	.nav__list-submenu li {
+		display: none;
+		flex-direction: column;
+		gap: 0.35rem;
```
Comment (nhcuongng) [inline src/components/pages/HomePage/NavBar/NavBar.scss]: Change to one unit

File: src/components/pages/HomePage/NavBar/NavBar.scss [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+		/* padding-right: 2px; Keep it center */
+		border: 1px solid black;
+		cursor: pointer;
+		pointer-events: auto;
+
+		transition: all 0.5s;
+	}
+
+	nav.closed .nav-toggle {
+		left: 333px;
```
Comment (nhcuongng) [inline src/components/pages/HomePage/NavBar/NavBar.scss]: change to percent unit

File: src/components/pages/HomePage/NavBar/NavBar.scss [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+		width: 80px;
+		height: 80px;
+	}
+
+	.nav-toggle.active {
+		left: 0;
+	}
+
+	nav.closed {
+		margin-left: -300px;
```
Comment (nhcuongng) [inline src/components/pages/HomePage/NavBar/NavBar.scss]: Change to left

File: src/components/pages/HomePage/NavBar/NavBar.scss [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+		left: 20px;
+		top: 33px;
+	}
+
+	.nav-toggle .arrow {
+		border-width: 0 3px 3px 0;
+	}
+
+	/* Mini version */
+	nav.mini {
```
Comment (nhcuongng) [inline src/components/pages/HomePage/NavBar/NavBar.scss]: Change to nested format

File: src/components/pages/HomePage/NavBar/NavSection/NavList/Index.jsx [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+export default function NavList({ items }) {
+	
+	console.log(items);
+
+	return (
+		<ul className="nav-list">
+			{items.map((i, iIndex) => {
+				return (
+					<NavItems
+						key={iIndex}
```
Comment (nhcuongng) [inline src/components/pages/HomePage/NavBar/NavSection/NavList/Index.jsx]: index only

File: src/components/pages/HomePage/NavBar/NavSection/NavList/NavItems/NavItemSubMenu.jsx [hunk 12]
```diff
@@ -0,0 +1,18 @@
+import React from "react"
+
+export default function NavItemSubMenu({ subMenu, isOpen }) {
+	
+	// console.log(subMenu);
+	// nav__list-submenu ${isOpen && "show"}}
+	return (
+		<ul className={`nav__list-submenu ${isOpen ? "show" : ""}`}>
+				{subMenu.map((subI, sIndex) => {
```
Comment (nhcuongng) [inline src/components/pages/HomePage/NavBar/NavSection/NavList/NavItems/NavItemSubMenu.jsx]: Change name var

File: src/components/pages/SignUpScreen/SignUpScreen.jsx [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+	faLock,
+	faUser,
+} from "@fortawesome/free-solid-svg-icons"
+import TextInputForm from "../../common/TextInputForm"
+import Button from "../../common/Button"
+import { Link } from "react-router-dom"
+
+export default function SignUpScreen() {
+	return (
+		<div className="sign-up">
```
Comment (nhcuongng) [inline src/components/pages/SignUpScreen/SignUpScreen.jsx]: "sign-up-screen"

File: src/App.jsx (hunk 2 above)
Comment (thomasdinh2k) [reply src/App.jsx:7]: routes has been transfer to main.jsx and re-setup as react-router-doom.v6 suggestion

File: src/components/common/Button.jsx (hunk 3 above)
Comment (thomasdinh2k) [reply src/components/common/Button.jsx]: styling for Button component has been encapsulated into a single stylesheet under its root folder
image

File: src/components/common/styles.scss (hunk 4 above)
Comment (thomasdinh2k) [reply src/components/common/styles.scss]: image

var has been encapsulated under _"src/styles/_variables.scss"_

File: src/components/pages/HomePage/NavBar/NavBar.scss (hunk 8 above)
Comment (thomasdinh2k) [reply src/components/pages/HomePage/NavBar/NavBar.scss]: switched to using SCSS variable for better comprehension

File: src/components/common/Button.jsx (hunk 3 above)
Comment (thomasdinh2k) [reply src/components/common/Button.jsx]: image

stylesheets have been encapsulated under **src/styles**

**Comment by thomasdinh2k** [comment]
6 commits have been pushed to **fix the issues addressed**. Please review again @nhcuongng

**Pull request merged**